File: vscode/src/cpp/colorDitection.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

using namespace std;

//画像ファイルの表示
void ditection(char* name)
{
	cv::Mat image = cv::imread(name);	//画像の読み込み
	cv::Mat dst,out;

	if (image.empty())	//画像が正しく読み込めたのかを確認
	{
		cout << "画像ファイルを読み込めませんでした．" << endl;
		return;
	}

    cvtColor(image,dst,CV_BGR2HSV);
    cv::inRange(dst,cv::Scalar(40,50,50),cv::Scalar(80,255,255),out);

	cv::imshow("元画像", image);	//画像の表示
    cv::imshow("HSV画像",dst);
    cv::imshow("結果画像",out);

    if (cv::waitKey() == 's')
	{
		cv::imwrite("../image/result/color-result.png",out);
	}

    cv::waitKey();


}

int main(int argc, char* argv[])
{
    if (argc < 2)
    {
        ditection("../image/angle1.png");
    }
	else{
        ditection(argv[1]);
    }
	
	return 0;
}

File: python/src/opencv/colorDitection.py
import sys
import cv2
import numpy as np

class colorDitection:

    def __init__(self,name):
        self.name = name

    def ditection(self):
        lower = np.array([40,50,50])
        upper = np.array([80,255,255])
        image = cv2.imread(self.name)
        hsv = cv2.cvtColor(image,cv2.COLOR_BGR2HSV)
        dst = cv2.inRange(hsv,lower,upper)

        cv2.imshow("元画像",image)
        cv2.imshow("HSV画像",hsv)
        cv2.imshow("結果画像",dst)

        if cv2.waitKey() == ord("s"):
            num = self.name.rfind("/")
            str = "./result/color-" + self.name[num+1:]
            cv2.imwrite(str,dst)

        cv2.waitKey(0)
        cv2.destroyAllWindows()

args = sys.argv
if len(args) > 1:
    cd = colorDitection(args[1])
else:
    cd = colorDitection("./image/angle1.png")

cd.ditection()